src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -1139,6 +1144,97 @@ void runtime::cpu::CPU_ExternalFunction::propagate_in_place_output(
     } while (propagate_further);
 }
 
+void runtime::cpu::CPU_ExternalFunction::process_in_place_concat(
```

Comment:
I think it would be much simpler if we where processing nodes in `post_order` (i.e. all users first) but it would be too much to work to change